pkg/crinit/aggregated/init.go

```diff
@@ -85,18 +89,25 @@ func newSubCmdInit(cmdOut io.Writer, pathOptions *clientcmd.PathOptions,
 			if err != nil {
 				glog.Fatalf("error: %v", err)
 			}
-
 			hostClientset, err := client.NewForConfig(hostConfig)
 			if err != nil {
 				glog.Fatalf("error: %v", err)
 			}
 
-			apiServiceClientset, err := apiregclient.NewForConfig(hostConfig)
+			aggConfig, err := util.GetClientConfig(pathOptions, opts.AggContext, opts.Kubeconfig).ClientConfig()
```

Comment:
I think the logic might be simpler to understand if you reused the hostClientset if the aggregator-context flag is not set, rather than creating a new clientset that points to the same host.